sdks/python/apache_beam/io/gcp/bigtableio.py

```diff
@@ -141,3 +147,197 @@ def expand(self, pvalue):
+          start_key = sample_row_key.row_key
+
+          suma = 0
+        suma += current_size
+        last_offset = sample_row_key.offset_bytes
+
+  def split_range_size(self, desired_size, sample_row_keys, range_):
+    start, end = None, None
+    l = 0
+    for sample_row in sample_row_keys:
+      current = sample_row.offset_bytes - l
+      if sample_row.row_key == b'':
+        continue
+
+      if(range_.start_key <= sample_row.row_key and
+         range_.end_key >= sample_row.row_key):
+        if start is not None:
+          end = sample_row.row_key
+          range_tracker = LexicographicKeyRangeTracker(start, end)
+
+          for split_key_range in self.split_range_subranges(current,
+                                                            desired_size,
+                                                            range_tracker):
+            yield split_key_range
+        start = sample_row.row_key
+      l = sample_row.offset_bytes
+
+  def range_split_fraction(self,
+                           current_size,
+                           desired_bundle_size,
+                           start_key,
+                           end_key):
+    range_tracker = LexicographicKeyRangeTracker(start_key, end_key)
+    return self.split_range_subranges(current_size,
+                                      desired_bundle_size,
+                                      range_tracker)
+
+  def fraction_to_position(self, position, range_start, range_stop):
+    return LexicographicKeyRangeTracker.fraction_to_position(position,
+                                                             range_start,
+                                                             range_stop)
+
+  def split_range_subranges(self,
+                            sample_size_bytes,
+                            desired_bundle_size,
+                            ranges):
+
+    last_key = copy.deepcopy(ranges.stop_position())
+    s = ranges.start_position()
+    e = ranges.stop_position()
+
+    split_ = float(desired_bundle_size) // float(sample_size_bytes)
+    split_count = int(math.ceil(sample_size_bytes // desired_bundle_size))
+
+    for i in range(split_count):
+      estimate_position = ((i + 1) * split_)
+      position = self.fraction_to_position(estimate_position,
+                                           ranges.start_position(),
+                                           ranges.stop_position())
+      e = position
+      yield iobase.SourceBundle(sample_size_bytes * split_, self, s, e)
+      s = position
+    if not s == last_key:
+      yield iobase.SourceBundle(sample_size_bytes * split_, self, s, last_key)
+
+  def read(self, range_tracker):
+    if range_tracker.start_position() is not None:
+      if not range_tracker.try_claim(range_tracker.start_position()):
+        # there needs to be a way to cancel the request.
+        return
+    start_position = range_tracker.start_position()
+    stop_position = range_tracker.stop_position()
```

Comment:
Consider adding setting a split points unclaimed callback here. This is not required but will make runners work better in some cases (for example, Dataflow autoscaling). See following for an example.
https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/avroio.py#L396

Reply:
I think I'm doing something wrong, because, I didn't find these functions in the LexicographicRangeTracker class.